### PR TITLE
WT-13007 Refactor compact tests

### PR DESCRIPTION
**Pull request by etienneptl**
- Creation of [compact_util.py](https://github.com/wiredtiger/wiredtiger/commit/d0e29ed763312f2562a2d664f514aef8121538c3) which contains methods used across all the compact tests
- Each test has been updated to use the base class

### CODE REVIEW

File: test/suite/compact_util.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+    def get_stat(self, stat, uri = None):
+        if not uri:
+            uri = ''
+        stat_cursor = self.session.open_cursor(f'statistics:{uri}', None, None)
+        val = stat_cursor[stat][2]
+        stat_cursor.close()
+        return val
+
+    def populate(self, uri, start_key, num_keys, value_size):
```
Comment (Sean04) [inline test/suite/compact_util.py:93]: If you add a value argument here you could replace the popoulation code in test_compact03.py:91

**Comment by Sean04** [comment]
`test_compact03.py` and `test_compact12.py` both do truncate so this could also be refactored into a util method. It's also likely to be used again in new tests.

File: test/suite/compact_util.py (hunk 1 above)
Comment (etienneptl) [reply test/suite/compact_util.py]: Nice, done in 5bd13c4fb35674183d71e3387c296fbbed876f21

**Comment by etienneptl** [comment]
> `test_compact03.py` and `test_compact12.py` both do truncate so this could also be refactored into a util method. It's also likely to be used again in new tests.

Good idea, done in b2c712f6f557c434f8441e0ebe0c46da574d852e

**Review by Sean04 — COMMENTED** [review]
`test_compact04` and `test_compact05.py` can also include some of the util functions.

File: test/suite/compact_util.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        val = stat_cursor[stat][2]
+        stat_cursor.close()
+        return val
+
+    def populate(self, uri, start_key, num_keys, value=None, value_size=None):
+        c = self.session.open_cursor(uri, None)
+        for k in range(start_key, num_keys):
+            if not value:
+                assert value_size is not None
+                c[k] = ('%07d' % k) + '_' + 'abcd' * ((value_size // 4) - 2)
```
Comment (Sean04) [inline test/suite/compact_util.py:98]:
```suggestion
                c[k] = ('%07d' % k) + '_' + 'a' * (value_size - 2)
```

just to simplify it

File: test/suite/compact_util.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+    def get_stat(self, stat, uri = None):
+        if not uri:
+            uri = ''
+        stat_cursor = self.session.open_cursor(f'statistics:{uri}', None, None)
+        val = stat_cursor[stat][2]
+        stat_cursor.close()
+        return val
+
+    def populate(self, uri, start_key, num_keys, value=None, value_size=None):
```
Comment (Sean04) [inline test/suite/compact_util.py:93]:
```suggestion
    def populate(self, uri, start_key, num_keys, value=None, value_size=1024):
```
Then you can remove the assert.

**Comment by etienneptl** [comment]
> `test_compact04` and `test_compact05.py` can also include some of the util functions.

You're right, see c42ac671b543989e281af4cece001609585eddbc.

File: test/suite/compact_util.py (hunk 3 above)
Comment (etienneptl) [reply test/suite/compact_util.py]: Good idea, see 8a97c9f96c8cebdc379a93429698defe653687cc

**Review by bzm-jas — APPROVED** [review]
LGTM:)

**Review by Sean04 — APPROVED** [review]
LGTM! Thanks for cleaning this up :)

**Pull request merged**